lang/java/reef-runtime-multi/src/main/java/org/apache/reef/runtime/multi/client/MultiRuntimeHDILocalDriverConfigurationProviderImpl.java

```diff
@@ -0,0 +1,83 @@
+import org.apache.reef.runtime.local.client.parameters.MaxNumberOfEvaluators;
+import org.apache.reef.runtime.local.client.parameters.RackNames;
+import org.apache.reef.runtime.local.driver.LocalDriverConfiguration;
+import org.apache.reef.tang.annotations.Parameter;
+import org.apache.reef.tang.formats.ConfigurationModule;
+
+import javax.inject.Inject;
+import java.net.URI;
+import java.util.Set;
+
+/**
+ * MultiRuntime configuration provider for HDI/Local runtime.
+ */
+public final class MultiRuntimeHDILocalDriverConfigurationProviderImpl extends AbstractDriverConfigurationProvider {
+  private final double jvmSlack;
+  private final int maxEvaluators;
+  private final Set<String> rackNames;
+
+  @Inject
+  private MultiRuntimeHDILocalDriverConfigurationProviderImpl(
+          @Parameter(JVMHeapSlack.class) final double jvmSlack,
+          @Parameter(MaxNumberOfEvaluators.class) final int maxEvaluators,
+          @Parameter(RackNames.class) final Set<String> rackNames) {
+    this.jvmSlack = jvmSlack;
+    this.maxEvaluators = maxEvaluators;
+    this.rackNames = rackNames;
+  }
+
+  @Override
+  protected String[] getDriverConfiguration(final URI jobFolder,
+                                            final String clientRemoteId,
+                                            final String jobId) {
+    final ConfigurationModule hdiModule = HDInsightDriverConfiguration.CONF
+            .set(HDInsightDriverConfiguration.JOB_IDENTIFIER, jobId)
+            .set(HDInsightDriverConfiguration.CLIENT_REMOTE_IDENTIFIER, clientRemoteId)
+            .set(HDInsightDriverConfiguration.JOB_SUBMISSION_DIRECTORY, jobFolder.toString())
+            .set(HDInsightDriverConfiguration.JVM_HEAP_SLACK, this.jvmSlack);
+    final String serializedHDIConfig = serializeConfiguration(hdiModule);
+    final String serializedHDIConfiguration = serializeRuntimeDefinition(
+            serializedHDIConfig,
+            true,
+            org.apache.reef.runtime.yarn.driver.RuntimeIdentifier.RUNTIME_NAME);
+
+    ConfigurationModule localModule = LocalDriverConfiguration.CONF
+            .set(LocalDriverConfiguration.MAX_NUMBER_OF_EVALUATORS, this.maxEvaluators)
+            .set(LocalDriverConfiguration.ROOT_FOLDER, ".")
+            .set(LocalDriverConfiguration.JVM_HEAP_SLACK, this.jvmSlack)
+            .set(LocalDriverConfiguration.CLIENT_REMOTE_IDENTIFIER, clientRemoteId)
+            .set(LocalDriverConfiguration.JOB_IDENTIFIER, jobId);
+    for (final String rackName : rackNames) {
+      localModule = localModule.set(LocalDriverConfiguration.RACK_NAMES, rackName);
+    }
+
+    final String serializedLocalConfig = serializeConfiguration(localModule);
+    final String serializedLocalConfiguration = serializeRuntimeDefinition(
+            serializedLocalConfig,
+            false,
+            org.apache.reef.runtime.local.driver.RuntimeIdentifier.RUNTIME_NAME);
+    return new String[]{serializedHDIConfiguration, serializedLocalConfiguration};
```

Comment:
I have a broader design question about this: Does this mean that multi-runtime only supports combinations of the local runtime with one other runtime? That falls a bit short of the potential of this change, IMHO.